libs/helm/autocomplete/src/lib/hlm-autocomplete.ts

```diff
@@ -0,0 +1,236 @@
+import { NgIcon, provideIcons } from '@ng-icons/core';
+import { lucideChevronDown, lucideSearch } from '@ng-icons/lucide';
+import { BrnCommandEmpty } from '@spartan-ng/brain/command';
+import { hlm } from '@spartan-ng/brain/core';
+import { BrnDialogState } from '@spartan-ng/brain/dialog';
+import { ChangeFn, TouchFn } from '@spartan-ng/brain/forms';
+import { BrnPopover, BrnPopoverContent } from '@spartan-ng/brain/popover';
+import {
+	HlmCommand,
+	HlmCommandEmpty,
+	HlmCommandGroup,
+	HlmCommandItem,
+	HlmCommandList,
+	HlmCommandSearch,
+	HlmCommandSearchInput,
+} from '@spartan-ng/helm/command';
+import { HlmIcon } from '@spartan-ng/helm/icon';
+import { HlmPopoverContent } from '@spartan-ng/helm/popover';
+import type { ClassValue } from 'clsx';
+import { HlmAutocompleteTrigger } from './hlm-autocomplete-trigger';
+
+export const HLM_AUTOCOMPLETE_VALUE_ACCESSOR = {
+	provide: NG_VALUE_ACCESSOR,
+	useExisting: forwardRef(() => HlmAutocomplete),
+	multi: true,
+};
+
+let nextId = 0;
+
+@Component({
+	selector: 'hlm-autocomplete',
+	imports: [
+		BrnPopover,
+		BrnPopoverContent,
+		HlmPopoverContent,
+
+		HlmCommand,
+		HlmCommandSearch,
+		HlmCommandSearchInput,
+		HlmCommandList,
+		HlmCommandGroup,
+		HlmCommandItem,
+		BrnCommandEmpty,
+		HlmCommandEmpty,
+
+		HlmAutocompleteTrigger,
+
+		NgIcon,
+		HlmIcon,
+	],
+	providers: [HLM_AUTOCOMPLETE_VALUE_ACCESSOR, provideIcons({ lucideSearch, lucideChevronDown })],
+	template: `
+		<brn-popover
+			align="start"
+			autoFocus="first-heading"
+			sideOffset="5"
+			closeDelay="100"
+			[closeOnOutsidePointerEvents]="true"
+			[state]="_popoverState()"
+			(stateChanged)="_stateChanged($event)"
+		>
+			<hlm-command
+				class="border-input focus-within:border-ring focus-within:ring-ring/50 h-9 border focus-within:ring-[3px]"
+			>
+				<hlm-command-search class="border-none" hlmAutocompleteTrigger [disabled]="!_filter()">
+					<ng-icon name="lucideSearch" hlm />
+					<input
+						#input
+						class="h-9 rounded-none py-1 text-base md:text-sm"
+						type="text"
+						autocomplete="off"
+						hlm-command-search-input
+						[id]="inputId()"
+						[placeholder]="placeholderText()"
+						[disabled]="_disabled()"
+						[value]="_filter()"
+						(input)="_onFilterChanged($event)"
+						(keydown.arrowDown)="_stateChanged('open')"
+					/>
+
+					<!-- TODO _toggleOptions and stateChanged interfere with each other  -->
+					<!-- TODO one click sets it popover state to open, second click triggers (stateChanged) with close because click outside and _toggleOptions state is set to open again -->
+					<button
+						class="flex items-center justify-center outline-none disabled:cursor-not-allowed [&>_ng-icon]:opacity-50"
+						tabindex="-1"
+						type="button"
+						aria-label="Toggle dropdown"
+						[disabled]="_disabled()"
+						(click)="_toggleOptions()"
+					>
+						<ng-icon name="lucideChevronDown" hlm />
+					</button>
```

Comment:
@ashley-hunter do you might have an idea on how to avoid this behavior?

Reply:
Sure, I think if we avoid trying to sync the state and instead get a view child of the popover and call the open and close methods on it directly, that should avoid this issue.

Reply:
Awesome idea! That works excellent